storage/time-series/influxdb/storer/app/App.js

```diff
@@ -0,0 +1,248 @@
+    // create callback to handle receive data
+    const boundInfluxWriter = this.influxWriter.writer.bind(this.influxWriter);
+    const callbackWriteData = async (tenant, deviceid, timestamp, attrs) => {
+      logger.debug('callbackWriteData: init');
+      logger.debug(`callbackWriteData: tenant=${tenant}`);
+      logger.debug(`callbackWriteData: deviceid=${deviceid}`);
+      logger.debug(`callbackWriteData: timestamp=${timestamp}`);
+      try {
+        await boundInfluxWriter(tenant, deviceid, attrs, timestamp);
+      } catch (e) {
+        logger.error('callbackWriteData:', e);
+      }
+    };
+
+    // create callback to handle delete device
+    const boundInfluxMeasurementDelete = this.influxMeasurement
+      .deleteMeasurement.bind(this.influxMeasurement);
+    const callbackDeleteMeasurement = async (tenant, deviceid) => {
+      logger.debug('callbackDeleteMeasurement: init');
+      logger.debug(`callbackDeleteMeasurement: tenant=${tenant}`);
+      logger.debug(`callbackDeleteMeasurement: deviceid=${deviceid}`);
+      try {
+        await boundInfluxMeasurementDelete(tenant, deviceid);
+      } catch (e) {
+        logger.error('callbackDeleteMeasurement:', e);
+      }
+    };
+
+    this.kafkaConsumer.registerCallbackDeviceData(
+      callbackWriteData,
+    );
+
+    this.kafkaConsumer.registerCallbackDeviceManager(
+      callbackWriteData,
+      configDelete['device.data.enable'] ? callbackDeleteMeasurement : null,
+    );
+  }
+
+  /**
+   * Associates callbacks with kafka tenant events to manipulation of influxdb organizations
+   */
+  initCallbacksTenantToCreateAndDelOrgs() {
+    // create callback to handle tenant create
+    const boundCreateOrgWithBucket = this.influxOrgs
+      .createOrgWithDefaultBucket.bind(this.influxOrgs);
+    const callbackCreateTenant = async (tenant) => {
+      logger.debug('callbackCreateTenant: init');
+      logger.debug(`callbackCreateTenant: tenant=${tenant}`);
+      try {
+        await boundCreateOrgWithBucket(tenant);
+      } catch (e) {
+        logger.error('callbackCreateTenant:', e);
+      }
+    };
+
+    // create callback to handle tenant delete
+    const boundWriterCloseOrg = this.influxWriter
+      .closeOne.bind(this.influxWriter);
+    const boundDeleteOrgWithBucket = this.influxOrgs
+      .deleteOrg.bind(this.influxOrgs);
+    const callbackDeleteTenant = async (tenant) => {
+      logger.debug('callbackDeleteTenant: init');
+      logger.debug(`callbackDeleteTenant: tenant=${tenant}`);
+      try {
+        await boundDeleteOrgWithBucket(tenant);
+        await boundWriterCloseOrg(tenant);
+      } catch (e) {
+        logger.error('callbackDeleteTenant:', e);
+      }
+    };
+
+    this.kafkaConsumer.registerCallbackTenants(
+      callbackCreateTenant,
+      configDelete['tenant.data.enable'] ? callbackDeleteTenant : null,
+    );
+  }
+
+  /**
+  * Defines the behaver for the shutdown
+  */
+  defineShutdown() {
```

Comment:
I think it is also necessary to close the connections with the Kafka and InfluxDB too.